src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -715,9 +717,9 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
                 diagnostics.Add(ErrorCode.ERR_RefReturningPropertiesCannotBeRequired, Location);
             }
 
-            if (IsAutoProperty)
+            if (IsAutoPropertyOrUsesFieldKeyword)
             {
-                if (!IsStatic && SetMethod is { IsInitOnly: false })
+                if (!IsStatic && ((_propertyFlags & Flags.HasAutoPropertySet) != 0) && SetMethod is { IsInitOnly: false })
```

Comment:
I'm not sure why we'd condition this on having `set;`, as opposed to any setter.

Reply:
Added an open question for language design: [https://github.com/dotnet/csharplang/pull/8389](https://github.com/dotnet/csharplang/pull/8389)